internal/shim-sev/layout.ld

```diff
@@ -0,0 +1,117 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+ENTRY(_start)
+EXTERN(__SALLYPORT_ABI_)
+
+PHDRS {
+    sallyport    PT_LOAD;
+
+    PML3         PT_LOAD;
+    PML4         PT_LOAD;
+
+    boot_text    PT_LOAD;
+    boot_data    PT_LOAD;
+
+    text         PT_LOAD;
+    rodata       PT_LOAD;
+    data         PT_LOAD;
+
+    dynamic      PT_DYNAMIC;
+
+    PT_ENARX_SALLYPORT  0x634A0001 FLAGS(0x0ff00000);
+    PT_ENARX_PML4       0x634A0002 FLAGS(0x0ff00000);
+    PT_ENARX_CODE       0x634A0003 FLAGS(0x0ff00000);
+}
+
+/* Configure this to a reasonable size
+   Current block size is 69632, which gives a maximum count of 15420
+   for the 3GB - 4GB range.
+*/
+_ENARX_SALLYPORT_BLOCK_COUNT = 5;
+_ENARX_SALLYPORT_BLOCK_SIZE = 69632;
```

Comment:
I might be misremembering. But I thought we had agreed on one PHDR sallyport entry for each separate thread.

Reply:
Also, it would be helpful for review to have the readelf output from the produced binary available. Perhaps even just a comment in the PR?

Reply:
> I might be misremembering. But I thought we had agreed on one PHDR sallyport entry for each separate thread.

In my model all CPUs/threads can use any free sallyport block. Allocating 512 blocks is waaaay too much and takes minutes on Milan to measure upfront. I would rather allocate additional blocks on demand, if a new CPU is started.

Reply:
> Also, it would be helpful for review to have the readelf output from the produced binary available. Perhaps even just a comment in the PR?

Added to the top comment

Reply:
It takes minutes using SEV on Milan. But we won't do that.

However, if we're going to commit to only starting one CPU up front, then we should only allocate one sallyport up front.

Reply:
2... one for debug print while filling out the other.

Reply:
changed